### DISCUSSION TITLE
sql

### DISCUSSION THREAD

**Discussion by PrabhakaranTankaey** [nextgenhealthcare/connect#4972] [Q&A]
Hi 
  I try to create global temp table in sql from mirth javascript.but it not work simultaneously
 first time it works and next time it not

This is my code:
	try	{
		var CreateQuery=DROP TABLE IF EXISTS ##testing_8209; CREATE TABLE ##testing_8209(AccountNumber nvarchar (2000) Null,AccountNumberone nvarchar (2000) Null)
		var dbConn = DatabaseConnectionFactory.createDatabaseConnection('com.microsoft.sqlserver.jdbc.SQLServerDriver',DbInsatnce,DbUser,DbPassword);
		var result = dbConn.executeCachedQuery(CreateQuery);	
		dbConn.close();
		}
	catch(e)
	{
		logger.info('CreateGlobalTempTable ='+ e.message);
		return e.message;
	}

**Comment by jonbartels** [comment]
Do you get any errors? 

Does it work if you break the statement into two calls? (See below)

Also what is the value or content of `result` after you call the query and it does not work? `result` may contain an error message from the DB server that would show the problem.

```
try {
var drop = "DROP TABLE IF EXISTS ##testing_8209;"
var create = "CREATE TABLE ##testing_8209 (AccountNumber nvarchar (2000) Null,AccountNumberone nvarchar (2000) Null)";
var dbConn = DatabaseConnectionFactory.createDatabaseConnection('com.microsoft.sqlserver.jdbc.SQLServerDriver',DbInsatnce,DbUser,DbPassword);
var result = dbConn.executeCachedQuery(drop);
var result = dbConn.executeCachedQuery(create);
dbConn.close();
}
catch(e)
{
logger.info('CreateGlobalTempTable ='+ e.message);
return e.message;
}
```

**Reply by PrabhakaranTankaey** [reply]
@jonbartels 
Hi,

  The  result returns "The statement did not return a result set."
After that i try to select table in ssms it throws  Invalid object name error.

Thanks in advance
